src/changes.ts

```diff
@@ -0,0 +1,103 @@
+
+/* IMPORT */
+
+import * as _ from 'lodash';
+import * as vscode from 'vscode';
+import Consts from './consts';
+import { toggleBox, boxNextLine } from './commands'
+import Document from './todo/document'
+import Utils from './utils';
+
+
+const Editor = {
+
+  textEditor: vscode.window.activeTextEditor
+
+}
+
+/* CHANGES */
+
+const Changes = {
+
+  textEditor: vscode.window.activeTextEditor,
+
+  doc: new Document ( Editor.textEditor ),
+
+  changes: [],
+  
+  lastNumberOfChanges: 0,
+
+  onChanges ({ document, contentChanges }) {
+
+    if ( document.languageId !== Consts.languageId ) return;
+
+    if ( !contentChanges.length ) return; //URL: https://github.com/Microsoft/vscode/issues/50344
+
+    Changes.changes.push ( ...contentChanges );
+
+    Changes.symbolInsertion ();
+
+  },
+
+  symbolInsertion () {
+    //Get the text of the last change
+    const changesLength = this.changes.length;
+    let lastChange = this.changes[ String( changesLength - 1 ) ],
+        currentLine = lastChange.range._end.line,
+        lastChangeText = lastChange.text;
+    
+    /*
+      Sometimes when you press an enter it makes 2 changes at the same time:
```

Comment:
This seems not very robust, what if VSC reports 3 or 4 changes? 🤔 Is that possible?

Also as I'll mention later I don't think we should listen for changes, we should perform something when "enter" or "shift+enter" are pressed, not when the content changes, right?